### DISCUSSION TITLE
Allowance for a critial humidity sensor

### DISCUSSION THREAD

**Discussion by NotExpectedYet** [TheRealWaldo/ha-optimal-humidity#175] [Ideas]
Hello! Firstly thank you for this integration. I'm currently controlling the morning condensation on my windows with your mold alert and optimal set points and it's been working well so far. My house has gone from an average of 60-70% humidity down to 40-55% which is amazing, the dehumidifiers are barely on as well. 

I was wondering if it would be possible to define a critial humidity sensor as well as temp, I gather this could replace the need for the calculation at the critical point. 

Most of my sensors are all temperature and humidity so makes sense to me to utilise a real sensor for that rather than a calculation. I'm no maths wizard, nor thermodynamics expert mind but I've noticed the calculation is generally above what I measure at the window.

**Comment by TheRealWaldo** [comment]
If you have a sensor at the critical point, I don't think you would need `ha-optimal-humidity` to avoid condensation or mold at that critical point.  Mold starts to form around 70%, and condensation at 100%.  You'd not want to turn on a humidifier if it exceeded that amount.  So you could use this in addition to your automation that considers the mold alert.

The comfortable humidity doesn't factor into the critical humidity except to avoid exceeding the mold (or condensation) level at the critical point by introducing humidity at the _not_ critical point.

This means it must be calculated value to figure out that 'the optimal humidity cause mold' and back it down, as it is, in essence, a prediction, not the actual value that matters.

**Comment by TheRealWaldo** [comment]
I guess I should ask @NotExpectedYet , are you also using my blueprint?  I could see this as a feature there.

**Reply by NotExpectedYet** [reply]
Hmmm interesting, I hadn't realised it was a cut off value for the the humidity at the critical point. I thought it was some deeper calculation but makes sense why it's not needed. Thank you for the thorough explanation. The optimal humidity calculation is working fine as a set point so I'll keep that and adjust my flow a bit. I can just trigger on the highest sensor value in the room for mold prevention. 

I'm not using it @TheRealWaldo, most of my automations are outside of HA. What's it for? I'd be interested to check it out, always looking for ideas for utilising this stuff more.

**Reply by NotExpectedYet** [reply]
I've found the blue prints, seems I'm inadvertently doing similar now after your comments. Wouldn't adding it there not be needed for similar reasons above or am I missing something?

**Reply by TheRealWaldo** [reply]
Essentially the `mold_warning` boolean does the same thing, as the calculated critical point relative humidity should be roughly the same as the actual.

Of course, fluid dynamics, location, the layout of the home, and other elements play a factor in the humidity at that location, so it's not 100% accurate.

Following the proposed automation strategy within Home Assistant, there is typically one automation per device.  In the case of controlling humidity, there are typically two devices.  A humidifier and a dehumidifier.  A dehumidifier could be an exhaust fan, an air conditioner, or a dedicated dehumidifier.

My blueprints consider the boolean to turn on dehumidifiers.

In my home, I don't always have an actual humidity sensor at the critical point in each room.  I never saw the need for the additional check because the calculation was close enough.

**Comment by palapapa** [comment]
A notification said I was mentioned here but I couldn't see it.

**Reply by NotExpectedYet** [reply]
It got reported it was spam, tagged me too
